src/core/LanguageController.ts

```diff
@@ -523,25 +525,36 @@ export default class LanguageController {
     applyTemplateData(sourceLanguageLines: string[], templateData: object) {
         let templateLines = [];
         for (const [templateKey, templateValue] of Object.entries(templateData)) {
-            //NOTE: this could be risky and end up removing var ${templateKey} from areas in the code where it is used for normal language operations
-            //We need to somehow split the bundle at a given point and only remove template variables above this point
-            const patterns = [
-                `var ${templateKey} =`, `var ${templateKey}=`, 
-                `const ${templateKey} =`, `const ${templateKey}=`,
-                `let ${templateKey} =`, `let ${templateKey}=`
-            ]
-            let p = 0
-            let index = -1
-            while(index == -1 && p < patterns.length) {
-                index = sourceLanguageLines.findIndex(element => element.includes(patterns[p]));
-                p++
-            }
-            
-            if (index != -1) {
-                sourceLanguageLines.splice(index, 1);
-            };
             templateLines.push(`var ${templateKey} = "${templateValue}";`);
         };
+
+        //Get lines in sourceLanguageLines which have ad4m-template-variable declared
+        const ad4mTemplatePattern = "//@ad4m-template-variable";
+        var indexes = [];
+        for(let i = 0; i < sourceLanguageLines.length; i++) {
+            if (sourceLanguageLines[i].includes(ad4mTemplatePattern)) {
+                indexes.push(i);
+            }
+        }
+        //Delete the line which is next to the comment 
+        for (let i = 0; i < indexes.length; i++) {
+            //Remove language variable comment
+            sourceLanguageLines.splice((indexes[i]) -i*2, 1);
+            //Remove template variable
+            sourceLanguageLines.splice((indexes[i]) -i*2, 1);
+        }
+
+        //@ts-ignore
+        if (templateData["dna"]) {
+            let dnaIndex = 0;
+            for(let i = 0; i < sourceLanguageLines.length; i++) {
+                if (sourceLanguageLines[i].includes(ad4mTemplatePattern)) {
+                    dnaIndex = i;
+                }
+            }
+            sourceLanguageLines.splice(dnaIndex, 1);
+        }
+
```

Comment:
Just resolved this with a much more robust system for templating that replaces the variables. Hopefully this will also remove the templating has issues we are seeing.